### PR TITLE
examples/DDS: Cleanup, remove sleep, etc.

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
We should get the DDS tests to the point, that they do not depend on "sleep" to synchronize state. So let's go the other way: Remove the sleeps to trigger them and let's fix them!

To make the output more readable, include some blank lines and reformat headers in a more consistant way.

---

Checklist:

* [X] Rebased against `dev` branch
* [X] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [X] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

**Comment by dennisklein** [comment]
I do not recommend adding sleeps at all. Why are they needed at all (the ones that are already present and are prolonged in this PR)?

**Comment by rbx** [comment]
I agree with Dennis. Remove all the sleeps. If then something doesn't work, open an issue in FairMQ SDK/fairmq-dds-command-ui/DDS.

**Comment by dennisklein** [comment]
Please note, that the CLI usage of `fairmq-dds-command-ui` has changed over time, see our canonical example for reference:
* FairMQ `v1.4.[8-12]`: https://github.com/FairRootGroup/FairMQ/blob/v1.4.8/examples/dds/fairmq-start-ex-dds.sh.in
* FairMQ `v1.4.13+`: https://github.com/FairRootGroup/FairMQ/blob/v1.4.13/examples/dds/fairmq-start-ex-dds.sh.in

**Comment by dennisklein** [comment]
Also note: https://github.com/FairRootGroup/FairMQ/issues/303

**Comment by ChristianTackeGSI** [comment]
The major issue is with the jun19 environment in FairSoft.
It has an old fairmq and dds version.
So I assume, that some races/etc have already been fixed in newer versions.

Hmmmm…

The alternative option is to ignore those tests in the jun19 case.
I'll look into this and into some other details.

**Comment by ChristianTackeGSI** [comment]
Thanks for the heads up!

I have rewritten this PR to go the other way: Let's find all the races!

So I have removed most of the sleeps. Should we remove the `sleep 2` also?

And to be able to read the output more easily, I have added a bit of empty lines, etc.

**Comment by ChristianTackeGSI** [comment]
Okay, dropping racy tests on FairSoft testsuite for jun19: https://github.com/FairRootGroup/FairSoft/pull/343

File: examples/MQ/pixelDetector/run/scripts/test-pixelSim.sh.in [hunk 1]
```diff
@@ -51,37 +51,44 @@ dds-topology -s $DDS_SESSION_ID --activate $FAIRMQ_DDS_TOPO_FILE --disable-valid
...
-fairmq-dds-command-ui -c t
-sleep 1
-fairmq-dds-command-ui -c d
-sleep 1
-fairmq-dds-command-ui -c q
-sleep 1
+for cmd in s t d q
+do
+	echo ""
+	echo "***** fairmq-dds-command-ui -c $cmd"
```
Comment (dennisklein) [inline examples/MQ/pixelDetector/run/scripts/test-pixelSim.sh.in:83]: Isn't `set -x` doing what you are you doing here?

File: examples/MQ/pixelDetector/run/scripts/test-pixelSim.sh.in [hunk 2]
```diff
@@ -51,37 +51,44 @@ dds-topology -s $DDS_SESSION_ID --activate $FAIRMQ_DDS_TOPO_FILE --disable-valid
...
+	echo ""
+	if (( ${#cmd} == 1 ))
+	then
+		echo "***** fairmq-dds-command-ui -c $cmd"
+		fairmq-dds-command-ui -c "$cmd"
+	else
+		echo "***** Waiting for state $cmd"
+		time fairmq-dds-command-ui -w "$cmd"
+	fi
+done
```
Comment (dennisklein) [inline examples/MQ/pixelDetector/run/scripts/test-pixelSim.sh.in:65]: In FairMQ < `1.4.13` this needs to be

```suggestion
fairmq-dds-command-ui -w "IDLE" -n ${requiredNofAgents}
fairmq-dds-command-ui -c i -w "INITIALIZING DEVICE" -n ${requiredNofAgents}
fairmq-dds-command-ui -c k -w "INITIALIZED" -n ${requiredNofAgents}
fairmq-dds-command-ui -c b -w "BOUND" -n ${requiredNofAgents}
fairmq-dds-command-ui -c x -w "DEVICE READY" -n ${requiredNofAgents}
fairmq-dds-command-ui -c j -w "READY" -n ${requiredNofAgents}
fairmq-dds-command-ui -c r
```

The `-n` argument expects the number of device addressed.

Comment (dennisklein) [reply examples/MQ/pixelDetector/run/scripts/test-pixelSim.sh.in:65]: See https://github.com/FairRootGroup/FairMQ/blob/93eb599df7d9ee4c869e2fe70c03b7a845739683/examples/dds/fairmq-start-ex-dds.sh.in#L57-L72

**Comment by dennisklein** [comment]
> So I have removed most of the sleeps. Should we remove the sleep 2 also?

This works only in DDS 2.4+
https://github.com/FairRootGroup/FairMQ/blob/93eb599df7d9ee4c869e2fe70c03b7a845739683/examples/dds/fairmq-start-ex-dds.sh.in#L33-L40

The commented lines work only in DDS 3.0+
https://github.com/FairRootGroup/FairMQ/blob/93eb599df7d9ee4c869e2fe70c03b7a845739683/examples/dds/fairmq-start-ex-dds.sh.in#L47-L52

**Comment by dennisklein** [comment]
I would just disable this example if there is no DDS `3.0+` and no FairMQ `1.4.13+` - otherwise you will have to write multiple versions of the script or clutter everything up with unreadable if-else cases which defeats the purpose as an example.

FairMQ 1.4.3 and DDS 2.2 are just too old and cumbersome.

File: examples/MQ/pixelDetector/run/scripts/test-pixelSim.sh.in (hunk 1 above)
Comment (ChristianTackeGSI) [reply examples/MQ/pixelDetector/run/scripts/test-pixelSim.sh.in:83]: Yes, `set -x` would do that. But do we want the whole other test logic to spam the logs?
What about rephrasing the output? Like "Sending command $cmd"?

**Comment by ChristianTackeGSI** [comment]
> I would just disable this example if there is no DDS `3.0+` and no FairMQ `1.4.13+` - otherwise you will have to write multiple versions of the script or clutter everything up with unreadable if-else cases which defeats the purpose as an example.

That sounds like a very good idea. If that's okay for everybody?

File: examples/MQ/pixelDetector/run/scripts/test-pixelSim.sh.in (hunk 1 above)
Comment (dennisklein) [reply examples/MQ/pixelDetector/run/scripts/test-pixelSim.sh.in:83]:
```
set -x
fairmq-dds-command-ui -c s
fairmq-dds-command-ui -c t
fairmq-dds-command-ui -c d
fairmq-dds-command-ui -c q
set +x
```
What about this?

I am honestly not sure if the loop is increasing readability here.

**Comment by dennisklein** [comment]
Once #1020 is green, let's (merge it and) rebase this PR on top. It should repair the macOS CI builds.

**Comment by dennisklein** [comment]
Please rebase on current `dev`

**Comment by ChristianTackeGSI** [comment]
Okay, I think, I'm done for now.
It's not the full blown everything upgraded PR, but the "one step after the other" one.

**Review by dennisklein — APPROVED** [review]

**Comment by karabowi** [comment]
I agree, I tried to write two different versions of the script, one for old DDS and FairMQ, and another one for the new versions.
It would be much cleaner to simply focus on the new one.

**Review by karabowi — APPROVED** [review]

**Pull request merged**